lib/include/qpx_mmc.h

```diff
@@ -1 +1,514 @@
+// some DVD related functions
+extern int get_rpc_state(drive_info *drive);
+extern int read_disc_regions(drive_info *drive);
+
+extern int css_disckey(drive_info *drive);
+extern int css_title(drive_info *drive, int32_t lba);
+// extern int css_titlekey( drive_info* drive, int lba, dvd_key_t p_title_key );
+// extern int css_unscramble( dvd_key_t p_key, uint8_t *p_sec );
+// extern void css_printkey (char *, uint8_t const * );
+
+// device capabilities detection functions
+extern int get_profiles_list(drive_info *drive);
+extern int get_features_list(drive_info *drive);
+extern int get_mode_pages_list(drive_info *drive);
+extern void detect_capabilities(drive_info *drive);
+
+// media information functions
+extern int read_atip(drive_info *drive, int silent);
+extern int read_toc(drive_info *drive, int silent);
+extern int read_track_info(drive_info *drive, trk *track, uint32_t track_n);
+extern int get_track_list(drive_info *drive);
+extern int read_disc_information(drive_info *drive);
+
+extern int read_capacities(drive_info *drive);
+extern int read_disc_info(drive_info *drive, int len);
+extern int determine_cd_type(drive_info *drive);
+// extern int determine_cdrw_subtype(drive_info* drive);
+extern int read_mediaid_dvd(drive_info *drive);
+extern int read_mediaid_dvdram(drive_info *drive);
+extern int read_mediaid_dvdminus(drive_info *drive);
+extern int read_mediaid_dvdplus(drive_info *drive);
+extern int read_mediaid_bd(drive_info *drive);
+extern int determine_disc_type(drive_info *drive);
+
+extern int mode_sense(drive_info *drive, int page, int page_control,
+                      int dest_len);
+extern int mode_select(drive_info *drive, int dest_len);
+
+// speed settings
+extern int get_spindown(drive_info *drive);
+extern int set_spindown(drive_info *drive);
+extern int get_performance(drive_info *drive, bool rw, uint8_t type);
+extern int get_write_speed_tbl(drive_info *drive);
+extern int set_streaming(drive_info *drive);
+extern int get_rw_speeds(drive_info *drive);
+extern int set_rw_speeds(drive_info *drive);
+extern int detect_speeds(drive_info *drive);
+
+// media change detection, media lock, load/eject
+extern int get_media_status(drive_info *drive);
+extern int start_stop(drive_info *drive, bool start);
+extern int load_eject(drive_info *drive, bool load, bool IMMED);
+extern int load_eject(drive_info *drive, bool IMMED);
+extern int get_lock(drive_info *drive);
+extern int set_lock(drive_info *drive);
+
+// extern void spinup(drive_info* drive);
+extern void spinup(drive_info *drive, uint8_t secs = 2);
+extern int seek(drive_info *drive, int32_t lba, uint8_t flags = 0);
+extern int play_audio_msf(drive_info *drive, msf beg, msf end);
+extern int play_audio(drive_info *drive, int32_t beg, short int len);
+extern int read_cd(drive_info *drive, uint8_t *data, int32_t lba,
+                   int sector_count, uint8_t flags, uint8_t FUA = 0);
+extern int read(drive_info *drive, uint8_t *data, int32_t lba, int sector_count,
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)